handler.lua

```diff
@@ -134,16 +149,25 @@ function CacheHandler:access(conf)
   local cached_val, err = red:get(cache_key)
   if cached_val and cached_val ~= ngx.null then
     ngx.log(ngx.NOTICE, "cache hit")
+    cache_status = 'HIT'
     local val = json_decode(cached_val)
     for k,v in pairs(val.headers) do
-      ngx.req.set_header(k, v)
+      ngx.header[k] = v
     end
-    return responses.send_HTTP_OK(val.content)
+    ngx.header[CACHE_HEADER] = cache_status
+    ngx.status = val.status_code
+    ngx.print(val.content)
+    return ngx.exit(val.status_code)
   end
 
   ngx.log(ngx.NOTICE, "cache miss")
+  cache_status = 'MISS'
+  ngx.ctx.rt_body_chunks = {}
+  ngx.ctx.rt_body_chunk_number = 1
+  ngx.header[CACHE_HEADER] = cache_status
```

Comment:
This should probably be a configurable header. The plugin schema should have a boolean value that determines whether to emit or not.

Reply:
works. I'll update the PR